cmd/vm-informant/main.go

```diff
@@ -64,3 +95,55 @@ func main() {
 		klog.Fatalf("Server failed: %s", err)
 	}
 }
+
+// runRestartOnFailure repeatedly calls this binary with the same flags, but with 'auto-restart'
+// removed.
+//
+// We execute ourselves as a subprocess so that it's possible to appropriately cleanup after
+// termination by various signals (or an unhandled panic!). This is worthwhile because we *really*
+// don't want to leave the cgroup frozen while waiting to restart.
+func runRestartOnFailure(args []string, cleanupHooks []func()) {
```

Comment:
what happens to signals and restarts?

Reply:
What do you mean?

Reply:
what happens if the informat gets a signal or gets killed? do we spiral with restarts? do we propogate signals to the child processes? by what mechanism? 

(and sort of relatedly, though not in my original comment) how do we distinguish inner/outer logging messages?)